### DISCUSSION TITLE
Why we need typescript?

### DISCUSSION THREAD

**Discussion by OmerAtayilmaz** [OmerAtayilmaz/learning-typescript#8] [Q&A] [answered]
Why we need typescript?
any idea 🤔

**Comment by RamazanYilmazz** [comment, marked as answer]
TypeScript is a programming language first developed by Microsoft in 2012. Its main ambition is to improve the productivity of developing complex applications.
It is an open-source language developed as a superset of JavaScript. What this means in simple terms is that any code valid in JavaScript is also valuable for TypeScript.
However, the language introduces optional features like typing or object-oriented programming. To benefit from these features, no library is required. Instead, just use the TypeScript compilation tool to transpile it (compiling the source code of one language into another language) into JavaScript. Thus, the executed code will be a Javascript equivalent of the compiled TypeScript code.  Did you understand?

**Reply by OmerAtayilmaz** [reply]
OH! ty its clear!
if I'm wrong pls warn me, TS used in compile time, not running time right?

**Reply by RamazanYilmazz** [reply]
TypeScript only performs static type checking at compile time! The generated JavaScript, which is what actually runs when you run your code, does not know anything about the types. you may consider this like javascript but its large sclae. For example JavaScript has 3 type but TypeSpring is has almost all of it. It's not a big deal.

**Reply by OmerAtayilmaz** [reply]
got it 💯. Ty for your helps.
